src/main/java/microsoft/exchange/webservices/data/core/ExchangeService.java

```diff
@@ -3928,58 +3932,51 @@ public boolean getExchange2007CompatibilityMode() {
   public void setExchange2007CompatibilityMode(boolean value) {
     this.exchange2007CompatibilityMode = value;
   }
-
+  
   /**
    * Retrieves the definitions of the specified server-side time zones.
    *
    * @param timeZoneIds the time zone ids
    * @return A Collection containing the definitions of the specified time
    * zones.
+ * @throws Exception 
    */
   public Collection<TimeZoneDefinition> getServerTimeZones(
-      Iterable<String> timeZoneIds) {
-    Date today = new Date();
+      Iterable<String> timeZoneIds) throws Exception {
+    Map<String, TimeZoneDefinition> timeZoneMap = new HashMap<String, TimeZoneDefinition>();
+    
+    GetServerTimeZonesRequest request = new GetServerTimeZonesRequest(this);
+	ServiceResponseCollection<GetServerTimeZonesResponse> responses = request.execute();
+	for (GetServerTimeZonesResponse response : responses) {
+		for (TimeZoneDefinition tzd : response.getTimeZones()) {
+			timeZoneMap.put(tzd.getId(), tzd);
+		}
+	}
+   
     Collection<TimeZoneDefinition> timeZoneList = new ArrayList<TimeZoneDefinition>();
+
     for (String timeZoneId : timeZoneIds) {
-      TimeZoneDefinition timeZoneDefinition = new TimeZoneDefinition();
-      timeZoneList.add(timeZoneDefinition);
-      TimeZone timeZone = TimeZone.getTimeZone(timeZoneId);
-      timeZoneDefinition.id = timeZone.getID();
-      timeZoneDefinition.name = timeZone.getDisplayName(timeZone
-          .inDaylightTime(today), TimeZone.LONG);
-			/*
-			 * String shortName =
-			 * timeZone.getDisplayName(timeZone.inDaylightTime(today),
-			 * TimeZone.SHORT); String longName =
-			 * timeZone.getDisplayName(timeZone.inDaylightTime(today),
-			 * TimeZone.LONG); int rawOffset = timeZone.getRawOffset(); int hour
-			 * = rawOffset / (60*60*1000); int min = Math.abs(rawOffset /
-			 * (60*1000)) % 60; boolean hasDST = timeZone.useDaylightTime();
-			 * boolean inDST = timeZone.inDaylightTime(today);
-			 */
+    	timeZoneList.add(timeZoneMap.get(timeZoneId));
     }
 
     return timeZoneList;
   }
-
+  
   /**
    * Retrieves the definitions of all server-side time zones.
    *
    * @return A Collection containing the definitions of the specified time
    * zones.
-   */
-  public Collection<TimeZoneDefinition> getServerTimeZones() {
-    Date today = new Date();
-    Collection<TimeZoneDefinition> timeZoneList = new ArrayList<TimeZoneDefinition>();
-    for (String timeZoneId : TimeZone.getAvailableIDs()) {
-      TimeZoneDefinition timeZoneDefinition = new TimeZoneDefinition();
-      timeZoneList.add(timeZoneDefinition);
-      TimeZone timeZone = TimeZone.getTimeZone(timeZoneId);
-      timeZoneDefinition.id = timeZone.getID();
-      timeZoneDefinition.name = timeZone.getDisplayName(timeZone
-          .inDaylightTime(today), TimeZone.LONG);
-    }
-
+ * @throws Exception 
+   */
+  public Collection<TimeZoneDefinition> getServerTimeZones() throws Exception {
```

Comment:
@serious6 The only reason - Collection was used in previous implementation. I didn't want to break API.